web/packages/teleport/src/Discover/Database/DeployService/AutoDeploy/SelectSecurityGroups.tsx

```diff
@@ -203,25 +203,22 @@ function withTips(
   securityGroups: SecurityGroup[],
   db: AwsRdsDatabase
 ): SecurityGroupWithRecommendation[] {
-  if (!db || !securityGroups) {
-    // return early without trying to add tips if there's no selected db or
-    // security groups given.
-    return securityGroups;
-  }
   const trustedGroups = getTrustedSecurityGroups(securityGroups, db);
-  return securityGroups.map(group => {
-    const isTrusted = trustedGroups.has(group.id);
-    const isOutboundAllowed = allowsOutbound(group);
-    return {
-      ...group,
-      tips: getTips(db, isTrusted, isOutboundAllowed),
-      // we recommend when either are true because security group rules are
-      // additive, meaning they can select multiple groups for a combined effect
-      // of satisfying the database inbound rules and the ECS task outbound
-      // rules.
-      recommended: isTrusted || isOutboundAllowed,
-    };
-  });
+  return (
+    securityGroups?.map(group => {
```

Comment:
not a frontend expert. what happens now when db is null?

Reply:
I removed the check at the top of `withTips` because I realized even if the db or its security groups are undefined, we can still highlight the security groups that allow outbound to the internet, as they don't depend on the db at all.

But if the db is undefined then it would be another bug... updated to fix that.